base/ca/src/main/java/com/netscape/cms/servlet/cert/EnrollServlet.java

```diff
@@ -480,23 +480,16 @@ private X509CertInfo[] handleCertAuthDual(X509CertInfo certInfo, AuthToken authT
                 "(&(x509cert.subject="
                         + certBasedOldSubjectDN + ")(!(x509cert.serialNumber=" + certBasedOldSerialNum
                         + "))(certStatus=VALID))";
-        CertRecordList list = cr.findCertRecordsInList(filter, null, 10);
-        int size = list.getSize();
-        Enumeration<CertRecord> en = list.getCertRecords(0, size - 1);
+        RecordPagedList<CertRecord> list = cr.findPagedCertRecords(filter, null, null);
 
-        logger.debug("EnrollServlet: signing cert filter " + filter);
+        logger.debug("EnrollServlet: signing cert filter {}", filter);
 
-        if (en.hasMoreElements()) {
-            X509CertInfo encCertInfo = new CertInfo();
-            X509CertInfo[] cInfoArray = new X509CertInfo[] { certInfo,
-                    encCertInfo };
-            int i = 1;
+        if (list.iterator().hasNext()) {
+            X509CertInfo encCertInfo;
+            X509CertInfo[] cInfoArray;
 
-            boolean encCertFound = false;
-
-            while (en.hasMoreElements()) {
-                CertRecord record = en.nextElement();
-                X509CertImpl cert = record.getCertificate();
+            for (CertRecord rec: list) {
```

Comment:
IIUC the loop will call `list.iterator()` implicitly, so the iterator object will be created twice. Does it matter?

Reply:
It is not a problem to generate a new iterator but looking at the code I think we can skip the if altogether.